### ISSUE TITLE
Add branch name to manually-started workflow runs

### ISSUE THREAD

**Issue by justinsmid** [refined-github/refined-github#6399]
### Description

(https://github.com/community/community/discussions/15690)

Currently, manually starting a workflow makes it so that workflow run does not show its branch in the actions overview:

![image](https://user-images.githubusercontent.com/34271675/222167869-67a944ed-159e-4152-9d60-7c77737cc6a4.png)

While workflows triggered automatically do:

![image](https://user-images.githubusercontent.com/34271675/222168656-e2f8821d-f2f4-4afa-88b7-e9204f204b34.png)

### Example URLs

https://github.com/refined-github/refined-github/actions

**Comment by fregante** [comment]
Indeed the branch does not show here, one of them was not on the default branch https://github.com/refined-github/sandbox/actions/workflows/Manual.yml

The request makes sense, although it might be a little niche compared to its complexity, for example we'd need to query every run to find the related branch, on every page load in the actions page, only for the rare occasion where this is useful: 

- user has manual dispatches
- user actually uses manual dispatches (a workflow could have the option to be manual, but never actually be called manually)
- user regularly dispatches from a non-default branch
- user actually cares about this distinction 
- user notices this utility


This pretty much leaves 3 users that would benefit from this feature 😅 

A userscript would probably be more beneficial here

**Comment by justinsmid** [comment]
Fair enough. I personally run into this issue quite often because we do deploys to staging and production manually and currently can't distinguish between the two easily since they're based on the branch that was used.

I can indeed definitely see it not being that common of an issue for others though, so I'll try to figure something out on our side!
